flycheck.el

```diff
@@ -45,6 +45,11 @@
   (require 'sh-script)      ; `sh-shell' for sh checker predicates
   (require 'jka-compr)      ; For JKA workarounds in `flycheck-temp-file-system'
   (require 'pcase)          ; `pcase-dolist' (`pcase' itself is autoloaded)
+  (if (featurep 'web-mode)
+      (require 'web-mode)
+    (defvar web-mode-engine nil)
+    (defvar web-mode-engine-file-regexps '()
+      "Engine file extensions."))
```

Comment:
This is a no-go.  Please don't introduce new `require`s or variable definitions from 3rd party packages.  Instead, use `boundp` at appropriate places to check whether the variables are defined and bound.